tests/e2e/pickdiff.spec.js

```diff
@@ -0,0 +1,115 @@
+// @ts-check
+const { test, expect } = require('@playwright/test');
+
+test.describe('PickDiff Application', () => {
+  test('should load the main page successfully', async ({ page }) => {
+    await page.goto('/');
+    
+    // Check that the page title is correct
+    await expect(page).toHaveTitle('PickDiff');
+    
+    // Check for main elements
+    await expect(page.locator('h1')).toContainText('PickDiff');
+    await expect(page.locator('#repo-path-container')).toBeVisible();
+    await expect(page.locator('#start-commit')).toBeVisible();
+    await expect(page.locator('#end-commit')).toBeVisible();
+    await expect(page.locator('#file-tree')).toBeVisible();
+  });
+
+  test('should display repository path', async ({ page }) => {
+    await page.goto('/');
+    
+    // Wait for repository path to load
+    await expect(page.locator('#repo-path')).not.toBeEmpty();
+    const repoPath = await page.locator('#repo-path').textContent();
+    expect(repoPath).toBeTruthy();
+  });
+
+  test('should load file tree', async ({ page }) => {
+    await page.goto('/');
+    
+    // Wait for file tree to populate
+    await page.waitForResponse(response => 
+      response.url().includes('/api/files') && response.status() === 200
+    );
+    
+    // Check that file tree has content
+    const fileTree = page.locator('#file-tree');
+    await expect(fileTree).toBeVisible();
+    
+    // The file tree should contain some files (at least this test file exists)
+    await expect(fileTree.locator('input[type="checkbox"]').first()).toBeVisible();
+  });
+
+  test('should validate form inputs before submission', async ({ page }) => {
+    await page.goto('/');
+    
+    // Wait for page to fully load
+    await page.waitForResponse(response => 
+      response.url().includes('/api/files') && response.status() === 200
+    );
+    
+    // Try to submit form without inputs
+    await page.click('button[type="submit"]');
+    
+    // Form should not submit without required fields
+    // Check that we're still on the same page (no navigation occurred)
+    expect(page.url()).toContain('localhost:3000');
```

Comment:
I'm not sure this assertion is working the way we expect. Even after inputting values into the fields and clicking "submit", the app remains on the same `localhost:3000` page.

Testing this use case manually, I can see that we get this popup if the user fails to fill the start/end commit fields:
<img width="453" height="116" alt="image" src="https://github.com/user-attachments/assets/cb9fd6c2-7300-453a-9dbe-c99451fd006d" />

So, maybe we need to make sure that this popup is shown instead of remaining on the same page.